sdk/src/Statechart/Helpers.hs

```diff
@@ -41,6 +42,13 @@ getAllTransitions NormalState{..} = transitions
 getAllTransitions MultiState{..} = transitions ++ concatMap getAllTransitions subStates
 getAllTransitions Parallel{..} = transitions ++ concatMap getAllTransitions regions
 
+lookupState :: Chart StateName e -> StateName -> Maybe (State StateName e)
+lookupState ch st =
+  foldr go Nothing (getAllChartStates ch)
+ where
+  go _ (Just x) = Just x
+  go a acc = if sid a == st then Just a else acc
```

Comment:
I think [this prelude function](https://hackage.haskell.org/package/prelude-compat-0.0.0.2/docs/Data-List2010.html#v:find) can replace most of `lookupState`